src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/WidgetPageObject.java

```diff
@@ -54,24 +77,38 @@ public WidgetPageObject navigate(String wikiUrl) {
     return this;
   }
 
-  public WidgetPageObject createAndNavigate(String wikiUrl) {
-    return create().navigate(wikiUrl);
-  }
-
-  public boolean isLoadedOnMercury() {
-    boolean result = isTagLoadedOnMercury();
+  public boolean isLoaded() {
+    boolean result = isWidgetVisible(getWidgetIFrame(), getWidgetBody());
     PageObjectLogging.log(getTagName(), MercuryMessages.VISIBLE_MSG, result);
     return result;
   }
 
-  public boolean isLoadedOnOasis() {
-    boolean result = isTagLoadedOnOasis();
-    PageObjectLogging.log(getTagName(), MercuryMessages.VISIBLE_MSG, result);
+  public boolean areAllValidSwappedForIFrames() {
+    boolean result = getWidgetWrapperList().size() == getWidgetIFrameList().size();
+    PageObjectLogging.log(getTagName(), MercuryMessages.ALL_VALID_WIDGETS_ARE_SWAPPED_MSG, result);
     return result;
   }
 
-  public WidgetPageObject createIncorrectAndNavigate(String wikiURL) {
-    return createIncorrect().navigate(wikiURL);
+  public boolean areLoaded() {
+    boolean result = true;
+    int i = 1;
+    List<WebElement> widgetIFrameList = getWidgetIFrameList();
+
+    if (widgetIFrameList.isEmpty()) {
+      result = false;
+    }
+
+    for (WebElement widgetIFrame: widgetIFrameList) {
+      if (!isWidgetVisible(widgetIFrame, getWidgetBody())) {
+        result = false;
+        PageObjectLogging.log(getTagName() + " #" + i, MercuryMessages.VISIBLE_MSG, result);
```

Comment:
I'm using `break` to exit the loop on first failure — not to check every tag if I already know the test fails. If I remove this log, script won't reach the log on the line 106 and the user won't know why the test failed. If you have an idea how to achieve this without code duplication, please share.